### PR TITLE
DEPR: Change numeric_only default to False in remaining groupby methods

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/internals/array_manager.py [hunk 1]
```diff
@@ -923,15 +923,13 @@ def idelete(self, indexer) -> ArrayManager:
     # --------------------------------------------------------------------
     # Array-wise Operation
 
-    def grouped_reduce(self: T, func: Callable, ignore_failures: bool = False) -> T:
```
Comment (jbrockmendel) [inline pandas/core/internals/array_manager.py:926]: i think you should be able to do the same in managers.py

Comment (rhshadrach) [reply pandas/core/internals/array_manager.py:926]: Indeed, thanks

**Review by mroeschke — APPROVED** [review]
LGTM merge when ready @jbrockmendel

File: pandas/core/groupby/groupby.py [hunk 2]
```diff
@@ -3772,18 +3638,15 @@ def blk_func(values: ArrayLike) -> ArrayLike:
...
         orig_mgr_len = len(mgr)
 
-        if numeric_only_bool:
+        if numeric_only:
             mgr = mgr.get_numeric_data()
 
-        res_mgr = mgr.grouped_reduce(blk_func, ignore_failures=False)
+        res_mgr = mgr.grouped_reduce(blk_func)
 
         if not is_ser and len(res_mgr.items) != orig_mgr_len:
```
Comment (jbrockmendel) [inline pandas/core/groupby/groupby.py:3646]: im pretty sure L3780-L3790 should be unnecesary now

Comment (jbrockmendel) [reply pandas/core/groupby/groupby.py:3646]: OK for follow-up.  lets merge this sucker

**Review by jbrockmendel — APPROVED** [review]

**Comment by jbrockmendel** [comment]
thanks @rhshadrach, exciting to see this go!  dont forget to update #30228

File: pandas/core/groupby/groupby.py (hunk 2 above)
Comment (rhshadrach) [reply pandas/core/groupby/groupby.py:3646]: It will never end, will it? :laughing: I'll put up a followup shortly.

**Comment by rhshadrach** [comment]
> dont forget to update #30228

Done.

File: pandas/core/groupby/groupby.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/groupby/groupby.py:3646]: ive got an "assorted cleanups" branch, will do it there

**Comment by jbrockmendel** [comment]
There are checkboxes for 40238 and 47561 that i'd expect to be checked off.  am i being too hopeful here?

**Comment by rhshadrach** [comment]
Nope - those are done as well. Thanks.

**Pull request merged**